pkg/prometheus/statefulset.go

```diff
@@ -58,36 +59,40 @@ var (
 	ManagedByOperatorLabels           = map[string]string{
 		managedByOperatorLabel: managedByOperatorLabelValue,
 	}
-	ShardLabelName                = "operator.prometheus.io/shard"
-	PrometheusNameLabelName       = "operator.prometheus.io/name"
-	ProbeTimeoutSeconds     int32 = 3
+	ShardLabelName                          = "operator.prometheus.io/shard"
+	PrometheusNameLabelName                 = "operator.prometheus.io/name"
+	PrometheusDeploymentKindLabeLName       = "operator.prometheus.io/deployment-kind"
```

Comment:
I'm not a big fan of `deployment-kind` (it feels a bit overloaded). Maybe `operator.prometheus.io/`mode with a value of either "server" or "agent".